UI/window-basic-auto-config.cpp

```diff
@@ -174,8 +174,16 @@ AutoConfigVideoPage::AutoConfigVideoPage(QWidget *parent)
 
 		// Calculate physical screen resolution based on the virtual screen resolution
 		// They might differ if scaling is enabled, e.g. for HiDPI screens
-		as_width = round(as_width * screen->devicePixelRatio());
-		as_height = round(as_height * screen->devicePixelRatio());
+		// NOTE: On Wayland, QScreen.size is physical rather than logical, so skip it
+
+#if !defined(__APPLE__) && !defined(_WIN32)
+		if (!QApplication::platformName().contains("wayland"))
```

Comment:
In order to properly support fractional scales under Wayland (for the purposes of figuring out the true screen resolution) do this:

1. Create a new protected function (stick it above `private:`) for these two classes. You want to override `QWidget::event` like this: `bool event(QEvent *event) override;`
2. Create a definition for that function, and write it something like this:

```cpp
bool OBSBasicSettings::event(QEvent *event)
{
	// When the DPR changes, we must recalculate the list since we depend on it's value
	if (event->type() == QEvent::DevicePixelRatioChange) {
		LoadResolutionLists();
	}
	return QWidget::event(event);
}
```
3. Ensure that you only access `devicePixelRatio` through the QWidget surface, like `window()->devicePixelRatio()`. Qt currently doesn't provide the true DPR on Wayland to any `QScreen()` AFAIK.

Note that this is inherently troublesome because it seems OBS will add every resolution for every screen? But you can only get the DPR of the screen this window is on. This is something you'll have to discuss with the maintainers though.

Reply:
> 3\. …Qt currently doesn't provide the true DPR on Wayland to any `QScreen()` AFAIK.

@redstrate, can we consider it a Qt bug ?

Reply:
> > 3. …Qt currently doesn't provide the true DPR on Wayland to any `QScreen()` AFAIK.
> 
> @redstrate, can we consider it a Qt bug ?

I would say yes, consider it a Qt bug (more specifically, in the Wayland QPA.) I personally don't know the feasibility of pushing the correct DPR up to QScreen (there are abandoned attempts like https://codereview.qt-project.org/c/qt/qtwayland/+/563344) but someone on the Qt bugtracker might.

Until such a thing happens though, we would still need the workaround described in my comment. Qt just did it to fix broken WebEngine fractional scaling see https://codereview.qt-project.org/c/qt/qtwebengine/+/569910

Reply:
Unfortunately, `LoadResolutionLists` is meant to be ran only once while creating the settings windows.
I don't think making it in an event is a good idea.

Reply:
Before I forgot, thanks for the feedback.